algolia/search/rule.go

```diff
@@ -66,6 +66,22 @@ func (r Rule) Equal(r2 Rule) bool {
 	return true
 }
 
+func (r Rule) MarshalJSON() ([]byte, error) {
+	type _Rule Rule
+	type _RuleWithOptionalCondition struct {
+		Condition *RuleCondition `json:"condition,omitempty"`
```

Comment:
I might be missing something here, but why don't we set `json:"condition,omitempty"` in the `Rule struct` directly? Instead of creating a tmp one.

https://github.com/algolia/algoliasearch-client-go/blob/9d7e6def70a5ea50700562d7651e7b572192eed6/algolia/search/rule.go#L14

Reply:
Sadly, `omitempty` only works on literal types, not user-defined structures. From the [Go documentation](https://golang.org/pkg/encoding/json/#Marshal):

> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.

Reply:
Oh too bad :(
All good then.